### PR TITLE
Show IP, Country, Data Center

### PR DESCRIPTION
**Pull request by SukkaW**
image

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -142,17 +208,29 @@ async function measureUpload(bytes, iterations) {
...
   );
 }
 
 async function speedTest() {
+  const serverLocationData = await fetchServerLocationData();
+
+  const { ip, loc, colo } = await fetchCfCdnCgiTrace();
+
+  logInfo('Your IP', ip);
+  logInfo('Your Country', loc);
```
Comment (KNawm) [inline cli.js:228]: How useful would be to know this? Maybe it can be added after the IP something like this: `192.168.0.0 (HK)`

File: cli.js [hunk 2]
```diff
@@ -142,17 +208,29 @@ async function measureUpload(bytes, iterations) {
...
+  logInfo('Server Location', `${serverLocationData[colo]} (${colo})`);
```
Comment (KNawm) [inline cli.js:229]: `Server Location` -> `Server location` to be consistent with the other texts

File: cli.js [hunk 3]
```diff
@@ -142,17 +208,29 @@ async function measureUpload(bytes, iterations) {
...
+    bold(' '.repeat(9 - size.length), size, 'speed:', yellow(`${speed} Mbps`))
   );
 }
 
 async function speedTest() {
+  const serverLocationData = await fetchServerLocationData();
+
+  const { ip, loc, colo } = await fetchCfCdnCgiTrace();
+
+  logInfo('Your IP', ip);
```
Comment (KNawm) [inline cli.js:227]: `Your IP` -> `Your network` to be consistent with CF, also it looks better IMO

File: cli.js (hunk 2 above)
Comment (KNawm) [inline cli.js:229]: Maybe move `serverLocationData[colo]` to a variable named `city` or something like that to improve readability

File: cli.js [hunk 4]
```diff
@@ -2,9 +2,75 @@
...
+}
+
+function fetchCfCdnCgiTrace() {
+  const parseCfCdnCgiTrace = (text) =>
+    text
+      .split('\n')
+      .map((i) => {
+        const j = i.split('=');
+
+        return [j[0].toLowerCase(), j[1]];
```
Comment (KNawm) [inline cli.js:58]: Isn't it already always lowercase?

File: cli.js (hunk 1 above)
Comment (SukkaW) [reply cli.js:228]: LGTM, I will change to that.

File: cli.js (hunk 3 above)
Comment (SukkaW) [reply cli.js:227]: Cloudflare use "Your Network" because it provided ASN information (which we don't have).

File: cli.js (hunk 2 above)
Comment (SukkaW) [reply cli.js:229]: LGTM, I will update to that.

**Comment by SukkaW** [comment]
image

File: cli.js (hunk 3 above)
Comment (KNawm) [reply cli.js]: Yeah it may be confusing

File: cli.js [hunk 5]
```diff
@@ -142,17 +207,30 @@ async function measureUpload(bytes, iterations) {
...
 
 async function speedTest() {
+  const serverLocationData = await fetchServerLocationData();
+
+  const { ip, loc, colo } = await fetchCfCdnCgiTrace();
+
+  const city = serverLocationData[colo];
+
+  logInfo('Your IP', `${ip} (${loc})`);
+  logInfo('Server location', `${city} (${colo})`);
```
Comment (KNawm) [inline cli.js:229]: Swap `Your IP` and `Server location` so the former appears above `Latency`, which is the same length, and all numbers will be grouped together.

**Comment by KNawm** [comment]
Implemented in [v2.0.0](https://www.npmjs.com/package/speed-cloudflare-cli/v/2.0.0) 🚀

**Pull request merged**